cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -164,6 +181,28 @@ private void doEnforce(EntityType entityType, EntityId parentId, Principal princ
     }
   }
 
+  /**
+   * Decrypt the given credential in encrypted and base64 encoded form.
+   */
+  private Principal decryptUserCredential(Principal principle) throws AccessException {
+    if (principle.getCredential() == null ||
+      !sConf.getBoolean(Constants.Security.Authentication.USER_CREDENTIAL_ENCRYPTION_ENABLED, false)) {
+      return principle;
+    }
+    try {
+      String plainCredential = new String(new TinkCipher(sConf).decryptFromBase64(principle.getCredential(),
```

Comment:
should we always create a new object `new TinkCipher` everytime method is called? OR it can be single object. From the constructor looks like just want to get keyset. may be we can have 1 object passed to the method?
Also  may be we should modify `decryptFromBase64 ` to return a string? Looks like thats how we want to use it.

Reply:
new TinkCipher throws exception, thus trying to avoid creating in constructor. 

Tink seems to be stateless and threadsafe. Could create a singleton. But would need to update the singleton whenever key in sConf changes. Let me run a benchmark and make this change in a follow up PR.